src/python/tools/config_generator.py

```diff
@@ -101,6 +102,18 @@ def update_stats(stats, config_dict):
     return config_dict
 
 
+def read_dataset_stats(path):
+    with open(path, "r") as f:
+        stats_dict = json.load(f)
+    
+    stats = list(stats_dict.values())
+    num_nodes_per_file = stats[2].strip('[]').split(", ")
+    num_nodes_per_file = [int(i) for i in num_nodes_per_file]
+    stats.pop(2)
+    stats = stats + num_nodes_per_file
+    return stats
```

Comment:
Change so this returns a dictionary that follows the json schema instead of a list.